py3status/parse_config.py

```diff
@@ -394,6 +396,28 @@ def make_value_from_env(self, param, value_type, function):
             self.notify_user("Environment variable `%s` undefined" % param)
         return self.value_convert(value, value_type)
 
+    def make_value_from_secret(self, param, value_type, function):
+        """
+        get parameter from secret-tool
+        """
+        param, value = self.remove_quotes(param), ""
+        module_name = self.current_module[-1].split()[0]
+        parameter = ".".join(["py3status", module_name, param])
+        command = "secret-tool lookup parameter {}".format(parameter)
```

Comment:
supporting `secret()` in `i3status.conf` — awesome 👍 But I don't feel good about promoting specifically `secret-tool`, in fact I haven't heard of it until now. There are other alternatives as well (most notably, `pass` and `gopass`), so I'm thinking, can we implement this so that we don't make any assumptions on what password manager people use, but instead require them to provide decryption command in py3status options?

For example, for `pass` the command would be `pass {} | head -n1`.

Reply:
FWIW, we can live without `secret()` because we can use `shell()` instead to use this on `secret-tool`, `pass`, `gopass` and others. The difference is that users would need to supply a long value... That and a questionable lack of `ast.literal_eval()`. 
```python
order += mail
imap {
    user = shell(secret-tool lookup parameter py3status.imap.user)
    password = shell(secret-tool lookup parameter py3status.imap.password)
}
```

I vaguely recall `shell()` was made for `pass` with or without `head -n1`. Try that?

Reply:
Ah I see, well in that case I'm not sure how much value there is to add `secret()`, especially because I doubt there are many people who have more than 3 secrets in their `i3status.conf`, so we aren't saving a lot of chars :)

`pass` prints out full contents, with password on the first line, but it's usual that people only store password there and nothing else, that's why calling `head -n1` is not always needed.